docs/deployment/configuration/resource_manager.rst

```diff
@@ -0,0 +1,113 @@
+The resource is allocated when the status is **“AllocationGranted”**. Qubole plugin calls:
+
+.. code-block:: go
+
+   status, err := AllocateResource(ctx, <cluster name>, <token string>, <constraint spec>)
+
+Wherein the placeholders are occupied by:
+
+.. code-block:: go
+
+   status, err := AllocateResource(ctx, "default_cluster", "flkgiwd13-akjdoe-0", ResourceConstraintsSpec{})
+
+The resource is deallocated when the Hive command completes its execution and the corresponding token is released. The plugin calls:
+
+.. code-block:: go
+
+   status, err := AllocateResource(ctx, <cluster name>, <token string>, <constraint spec>)
+
+Wherein the placeholders are occupied by:
+
+.. code-block:: go
+
+   err := ReleaseResource(ctx, "default_cluster", "flkgiwd13-akjdoe-0")
+
+Below is an example interface that shows allocation and deallocation of resources.
+
+.. code-block:: go
+
+    type ResourceManager interface {
+    GetID() string
+    // During execution, the plugin calls AllocateResource() to register a token in the token pool associated with a resource
+    // If it is granted an allocation, the token is recorded in the token pool until the same plugin releases it.
+    // When calling AllocateResource, the plugin has to specify a ResourceConstraintsSpec that contains resource capping constraints at different project and namespace levels.
+    // The ResourceConstraint pointers in ResourceConstraintsSpec can be set to nil to not have a constraint at that level
+    AllocateResource(ctx context.Context, namespace ResourceNamespace, allocationToken string, constraintsSpec ResourceConstraintsSpec) (AllocationStatus, error)
+    // During execution, after an outstanding request is completed, the plugin uses ReleaseResource() to release the allocation of the token from the token pool. This way, it redeems the quota taken by the token
+    ReleaseResource(ctx context.Context, namespace ResourceNamespace, allocationToken string) error
+    }
+
+How can you force ResourceManager to force runtime quota allocation constraints?
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Runtime quota allocation constraints can be achieved using ResourceConstraintsSpec. It is a contact that a plugin can specify at different project and namespace levels.
+
+Let's take an example to understand it.
+
+You can set ResourceConstraintsSpec to ``nil`` objects, which means there would be no allocation constraints at the respective project and namespace level. When ResourceConstraintsSpec specifies ``nil`` ProjectScopeResourceConstraint, and a non-nil NamespaceScopeResourceConstraint, it suggests no constraints specified at any project or namespace level.
```

Comment:
```suggestion
For example, you can set ResourceConstraintsSpec to ``nil`` objects, which means there would be no allocation constraints at the respective project and namespace level. When ResourceConstraintsSpec specifies ``nil`` ProjectScopeResourceConstraint, and a non-nil NamespaceScopeResourceConstraint, it suggests no constraints specified at any project or namespace level.
```

Reply:
I'm a little confused by this example -- wouldn't a non-nil NamespaceScopeResourceConstraint suggest a constraint at the namespace level?